decidim-elections/app/commands/decidim/votings/admin/manage_polling_station.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Votings
+    module Admin
+      # A command with the shared business logic to manage a polling station
+      class ManagePollingStation < Rectify::Command
+        def manage_polling_officers(polling_station, president_id, managers_ids)
+          manage_president(polling_station, president_id)
+          manage_managers(polling_station, managers_ids)
+        end
+
+        private
+
+        def manage_president(polling_station, president_id)
+          unassign_president(polling_station)
+          return if president_id.nil?
+
+          assign_president(president_id, polling_station)
+        end
+
+        def unassign_president(polling_station)
+          assign_president(polling_station.polling_station_president.id, nil) if polling_station.polling_station_president.present?
+        end
+
+        def assign_president(president_id, polling_station)
+          polling_station_president = PollingOfficer.find_by(id: president_id)
+          polling_station_president.presided_polling_station = polling_station
+          polling_station_president.save!
+        end
+
+        def manage_managers(polling_station, managers_ids)
+          unassign_all_managers(polling_station)
+          assign_new_managers(polling_station, managers_ids)
+        end
+
+        def unassign_all_managers(polling_station)
+          polling_station.polling_station_managers.each do |manager|
+            manager.managed_polling_station = nil
+            manager.save!
+          end
```

Comment:
What about `polling_station.polling_station_managers.update_all(managed_polling_station: nil)`?

Reply:
Oooooooh shortcuts 😍

Reply:
Rubocop does not like `update_all` because it updates the DB directly skipping `ActiveRecord`'s validations (https://stackoverflow.com/a/39646223).
Since in real life we shouldn't have more than a couple of managers, this is not really a performance concern and I'll follow `rubocop`'s reccommendation.